packages/sitecore-jss-react/src/enhancers/withSitecoreContext.tsx

```diff
@@ -38,3 +38,28 @@ export function withSitecoreContext(options?: WithSitecoreContextOptions) {
 
   };
 }
+
+/**
+ * This hook grants acсess to the current SiteCore page context
+ * by default JSS includes the following properties in this context:
+ * - pageEditing - Provided by Layout Service, a boolean indicating whether the route is being accessed via the Experience Editor.
+ * - pageState - Like pageEditing, but a string: normal, preview or edit.
+ * - site - Provided by Layout Service, an object containing the name of the current Sitecore site context.
+ *
+ * @see https://jss.sitecore.com/docs/techniques/extending-layout-service/layoutservice-extending-context
+ *
+ * @example
+ * const EditMode = () => {
+ *    const {pageEditing} = useSitecoreContext();
+ *    return <span>Edit Mode is {pageEditing ? 'active' : 'inactive'}</span>
+ * }
+ */
+export function useSitecoreContext(options?: WithSitecoreContextOptions) {
+  const reactContext = React.useContext(SitecoreContextReactContext);
+  const updatable = options?.updatable;
+
+  return {
```

Comment:
With the addition of `updatable` functionality, this no longer matches the signature expected in the @example above (or original PR). Either need to update the example, or (preferably) simplify the return to match. How about:
```
return {
	...reactContext.context,
	setContext: updatable ? reactContext.setContext : undefined
}
```
So that context properties are available at the root? Maybe add a second example above with use of updatable / setContext.

Reply:
@ambrauer We should have consistence between withSitecoreContext and useSitecoreContext.
withSitecoreContext provides `sitecoreContext` and `updateSitecoreContext`.
Updated example

Reply:
@sc-illiakovalenko Agree, the consistency here is more important.